controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -747,46 +707,13 @@ func applyClusterTemplateAndDatacenterTemplateConfigs(t *testing.T, ctx context.
 	assert.Equal(dc2Size, dc2.Spec.Size)
 	assert.Equal(*dc2.Spec.CDC.PulsarServiceUrl, "pulsar://test-url")
 
-	actualConfig, err = gabs.ParseJSON(dc2.Spec.Config)
-	require.NoError(err, fmt.Sprintf("failed to parse dc2 config %s", dc2.Spec.Config))
-
-	expectedConfig, err = parseCassandraConfig(kc.Spec.Cassandra.Datacenters[1].DatacenterOptions.CassandraConfig, serverVersion, 3, "dc1", "dc2")
-	require.NoError(err, "failed to parse CassandraConfig")
-	assert.Equal(expectedConfig, actualConfig)
-
 	t.Log("deleting K8ssandraCluster")
 	err = f.DeleteK8ssandraCluster(ctx, client.ObjectKey{Namespace: kc.Namespace, Name: kc.Name}, timeout, interval)
 	require.NoError(err, "failed to delete K8ssandraCluster")
 	f.AssertObjectDoesNotExist(ctx, t, dc1Key, &cassdcapi.CassandraDatacenter{}, timeout, interval)
 	f.AssertObjectDoesNotExist(ctx, t, dc2Key, &cassdcapi.CassandraDatacenter{}, timeout, interval)
 }
 
-func parseCassandraConfig(config *api.CassandraConfig, serverVersion string, systemRF int, dcNames ...string) (*gabs.Container, error) {
```

Comment:
This is the function that was being used to test serialization of config by mimicking the behavior of the actual serialization function.